distributed/scheduler.py

```diff
@@ -4015,31 +4164,136 @@ def __init__(
         self.rpc.allow_offload = False
         self.status = Status.undefined
 
+    ####################
+    # state properties #
```

Comment:
I would prefer us to be much more conservative with API exposure. This PR already removes a few underscores and I would rather see the entire state to be closed off unless there is any concrete reason for it to be public.